cat/spirv.cat

```diff
@@ -0,0 +1,127 @@
+SPIRV
+
+(* Terms from spirv.als *)
+// stor = [] "set to relation, (s,s) for all s in E, lift a set to a relation (the [_] operator in Herd)"
+// rai-r = r | r^-1
+// sthd = int
+// immpo = po - po;po
+// rc[r] = r?
+// ssg, swg, sqf = same scope id @ subgroup, workgroup, queuefamily
+// inscope = sr
+// sloc = loc
+// sref = vloc
+// scbarinst = syncbar "relates pairs of control barriers, same control barrier dynamic instance"
+// imm-r = r - (r;r+)
+// twoplus-r = r;r+
+
+//???????????
+// chains = new relation
+// ssw = new relation for system-synchronizes-with?
+
+(*******************)
+(*  Auxiliaries *)
+(*******************)
+let mutordatom = [ATOM]; ((loc & vloc & sr) \ id); [ATOM] // mutually ordered atomics (same loc, inscope)
+
+let sctosem = SC0 * SEMSC0 | SC1 * SEMSC1
+let semtosc = SEMSC0 * SC0 | SEMSC1 * SC1
+
+let avvisSem = ((SC0 | SC1) * ((AV | VIS) & DV)) | (sctosem; [SEMAV | SEMVIS])
+let avvisSet = [AV | VIS]; (vloc & loc)
+let avvisinc = (avvisSem | avvisSem^-1) \ id |
+               (avvisSet | avvisSet^-1)
+
+let posctosem = sctosem & po
+let posemtosc = semtosc & po
+
+let asmo = [ATOM & W]; mutordatom; [ATOM & W]
```

Comment:
In the allow model, `asmo` is supposed to be base relation having pretty much the role of `co`. I think the definition should be `asmo = co & (Atom * Atom)`

Reply:
Updated. 
With mutordatom-atom-W pairs as coherence pair, was the restirction line still needed @line110?

Reply:
I don't understand what the question is. This axiom is similar to the `axiom-Coherence` in PTX, so I do believe we need it

Reply:
Okay I was misunderstanding the mutordatom with asmo.